DEV_GUIDE.md

```diff
@@ -0,0 +1,102 @@
+| `-DskipITs=true`   | skip integration tests                                                                    |
+| `-DskipTests=true` | skip all tests                                                                            |
+| `-Pdebug`          | enables logging so you can see what the Kafka clients, Proxy and in VM brokers are up to. |
+
+The kafka environment used by the integrations tests can be _defaulted_ with these two environment variables.
+
+| env var                       | default | description                                                                                                                             |
+|-------------------------------|---------|-----------------------------------------------------------------------------------------------------------------------------------------|
+| `TEST_CLUSTER_EXECUTION_MODE` | `IN_VM` | `IN_VM` or `CONTAINER`. if `IN_VM`, kafka will be run same virtual machines as the integration test. Otherwise containers will be used. |
+| `TEST_CLUSTER_KRAFT_MODE`     | `true`  | if true, kafka will be run in kraft mode.                                                                                               |
+
+When the integration-tests are run in `CONTAINER` mode, the kafka/zookeeper logs are written to a location specified by
+the `container.logs.dir`  system property. When run through Maven this is defaulted to `integrationtests/target/container-logs`.
+
+Pass the `-Dquick` option to skip all tests and non-essential plug-ins and create the output artifact as quickly as possible:
+
+```shell
+$ mvn clean verify -Dquick
+```
+
+### Code Formatting
+
+Run the following command to format the source code and organize the imports as per the project's conventions:
+
+```shell
+$ mvn process-sources
+```
+
+## Running Integration Tests on Podman
+
+There is an incompatibility between HTTP connection timeout expectations of 
+[testcontainers-java](https://github.com/testcontainers/testcontainers-java) and the Podman API. This
+can result in sporadic test failures when running the Integration Tests under Podman.  It manifests as
+failed or hanging REST API calls that lead to test failures and test hangs.
+
+It affects Linux and Mac OS X.
+
+To workaround around the issue, turn off the `service_timeout` by following these instructions.
+
+### MacOS X
+
+Start the `podman` machine as normal, then:
+
+```shell
+echo 'mkdir -p /etc/containers/containers.conf.d && printf "[engine]\nservice_timeout=0\n" > /etc/containers/containers.conf.d/service-timeout.conf' |  podman machine ssh --username root --
+```
+
+### Linux
+
+As a privileged user:
+
+```shell
+mkdir -p /etc/containers/containers.conf.d && printf "[engine]\nservice_timeout=0\n" > /etc/containers/containers.conf.d/service-timeout.conf
+```
+
+### Verify that the fix is effective
+
+Start this command:
+```shell
+socat - UNIX-CONNECT:/var/run/docker.sock
```

Comment:
on my machine that's docker's sock (I have both installed) and `podman.sock` is at `/run/user/1000/podman/podman.sock`

Reply:
With podman machine, you get /var/run/docker.sock.
Guess I should use $(podman info --format '{{.Host.RemoteSocket.Path}}') in the verification command.